### ISSUE TITLE
Incorrect incorrect_ref_pattern rule?

### ISSUE THREAD

**Issue by Ilyes512** [IBM/openapi-validator#301]
So I have a pretty large openapi yaml file that I split up into multiple files for organisational purposes. Currently I am getting allot of warnings that I previously with an older version (version 0.38 to 0.46) did not get. As far as I understand it the warning or not correct. For instance I get:

```
  Message :   schema $refs must follow this format: *#/components/schemas*
  Rule    :   incorrect_ref_pattern
  Path    :   paths./foobar/acme.get.responses.200.content.application/json.schema.allOf.0.$ref
  Line    :   56
```
The ref it looks like this:

```
- $ref: "schemas/foobar.yml#/components/schemas/FoobarResourcesDocument"
```

Is this a false positive or am I wrong? Or maybe its not even supported? (OpenApi doc says it is: https://swagger.io/docs/specification/using-ref/).

**Comment by dpopp07** [comment]
Hey @Ilyes512, thanks for the issue. This does look like a false positive to me. I'm guessing that check is getting confused by the multi-file setup. Though, this isn't a new check, nor have we seen this before. I'll try to look into it when I get a chance.

**Comment by barrett-schonefeld** [comment]
Yes, I think the problem here is that your document is likely being treated as a Swagger 2 document rather than the appropriate OAS3 document.

The way to resolve this would be to be more sophisticated in the validator about how we detect the `OpenAPI` version or to explicitly include the `OpenAPI` version in the document.

**Comment by Ilyes512** [comment]
The root file and every file that I reference always starts with:

```yaml
---
openapi: 3.0.3
# ... the rest
```

Latest version is `3.1.0`. Not sure if updating the version to the latest will help?

**Comment by barrett-schonefeld** [comment]
Looking into this now.

I created a test case with the offending `$ref`, and I am not able to reproduce the error.

If you could provide the context in which the `$ref` is used, I may be able to replicate the behavior more closely in the test case to see if I get the error.

**Comment by barrett-schonefeld** [comment]
I opened PR #321, which provides a test case meant to mimic the error outlined in the initial issue.

**Comment by Krapaince** [comment]
Hey @barrett-schonefeld, I ran into the same issue with  a multiple files openapi documentation.
```
  Message :   schema $refs must follow this format: *#/components/schemas*
  Rule    :   incorrect_ref_pattern
  Path    :   paths./user/login.post.requestBody.content.application/json.schema.$ref
  Line    :   18

  Message :   schema $refs must follow this format: *#/components/schemas*
  Rule    :   incorrect_ref_pattern
  Path    :   paths./user/login.post.responses.200.content.application/json.schema.$ref
  Line    :   18

  Message :   schema $refs must follow this format: *#/components/schemas*
  Rule    :   incorrect_ref_pattern
  Path    :   paths./user/login.post.responses.401.content.application/json.schema.$ref
  Line    :   18
  ```
> I created a test case with the offending `$ref`, and I am not able to reproduce the error.

I created a [repo](https://github.com/Krapaince/ibm-openapi-validator-incorrect_ref_pattern) so that you can reproduce the error hopefully. Otherwise, have a look at the action's [log](https://github.com/Krapaince/ibm-openapi-validator-incorrect_ref_pattern/runs/6064425793?check_suite_focus=true#step:5:19)

**Comment by Krapaince** [comment]
I have done some tests with the validator and it seems that with multiple `paths`, the validator reports errors where there isn't see https://github.com/IBM/openapi-validator/issues/301#issuecomment-1101507533.
Whereas when there is only one `path` enabled  (see [14fbc95]( https://github.com/Krapaince/ibm-openapi-validator-incorrect_ref_pattern/tree/14fbc952adc85365574099044f752c93641da729)), no false positive warning is output ([log](https://github.com/Krapaince/ibm-openapi-validator-incorrect_ref_pattern/runs/6069030888?check_suite_focus=true#step:5:9)).

**Comment by dpopp07** [comment]
@Krapaince Sorry for losing track of this - I'll take a look at your test repo soon

**Comment by dpopp07** [comment]
@Krapaince I believe this has been resolved in the validator since this issue was opened (version 0.75.0), as we completely re-did the ref pattern rule when converting it into the Spectral format. I was unable to reproduce with your test repo. Feel free to re-open if you're still seeing the issue
